crawlerapp/app/pageparse.js

```diff
@@ -5,27 +5,53 @@ var request = require('request');
 
 module.exports = {
 
-    searcHelper: function(url, searchType, searchDepth, keyword) {
-        //TEST VARIABLES
-        searchType = 1;
-        searchDepth = 1;
-        keyword = "";
-
-        if (urlIsValid(url)) {
-            //TODO: CALL APPROPRIATE CRAWL BASED ON TYPE
-            getPage(url);
+    searchHelper: function(url, searchType, searchDepth, keyword) {
+        if (searchType == 0) {
+            return this.crawlDepthFirstHelper(url, searchDepth, keyword);
         } else {
-            console.log("Invalid URL\n");
-            return false;
+            console.log("BFS NOT IMPLEMENTED!\n");
+            return this.crawlBreadthFirstHelper(url, searchDepth, keyword);
         }
-    },
-    crawlDepthFirst: function() {
 
     },
-    crawlBreadthFirst: function() {
+    crawlDepthFirstHelper: function(url, searchDepth, keyword) {
+        var crawlRes = this.crawlDepthFirst(url, searchDepth, 0, keyword);
+        return crawlRes;
+    },
+    crawlDepthFirst: function(url, searchDepth, currentDepth, keyword) {
+        return puppeteer.launch()
+            .then(async browser => {
+                var page = await initBrowser(browser);
+                var html = await navigateUrl(page, url);
+                browser.close();    //close now to save memory
+
+                var newDepth = currentDepth + 1;
+                var crawlRes = {"url": url,
+                        "depth": currentDepth};
+
+                var urls = parseHtml(url, html);
+                crawlRes.links = urls;
+
+                var found = await findKeyword(html, keyword);
+                crawlRes.keywordFound = found;
+
+                var chosenUrl = await chooseRandomUrl(urls);
+
+                if (currentDepth < searchDepth && found == false) {
+                    crawlRes.links[chosenUrl] = await this.crawlDepthFirst(chosenUrl, searchDepth, newDepth, keyword);
+                    return crawlRes;
+                } else {
+                    return crawlRes;
+                }
+            })
+            .catch(function(err) {
+                console.log(err);
+            });
+    },
+    crawlBreadthFirstHelper: function() {
 
     },
-    findKeyword: function() {
+    crawlBreadthFirst: function() {
 
```

Comment:
Not sure if this will cause issues on the FE if the user makes a request for a bfs from the form. Maybe we could just return a mock json object with static values saying not implemented so the FE can still display something while we are still getting this function written.